code/game/machinery/computer/abnormality_ego.dm

```diff
@@ -56,13 +57,42 @@
 				to_chat(usr, "<span class='warning'>Not enough PE boxes stored for this operation.</span>")
 				playsound(get_turf(src), 'sound/machines/terminal_prompt_deny.ogg', 50, TRUE)
 				return FALSE
-			var/obj/item/I = new E.item_path(get_turf(src))
-			A.stored_boxes -= E.cost
-			A.current_ego += I
-			to_chat(usr, "<span class='notice'>[I] has been dispensed!</span>")
-			playsound(get_turf(src), 'sound/machines/terminal_prompt_confirm.ogg', 50, TRUE)
-			updateUsrDialog()
-			return TRUE
+			if(active)
+				to_chat(usr, "<span class='warning'>Console is already extracting.</span>")
+				playsound(get_turf(src), 'sound/machines/terminal_prompt_deny.ogg', 50, TRUE)
+				return FALSE
+
+			active = TRUE
+			//Some roles get faster extraction
+			var/user_role = usr?.mind?.assigned_role
+			var/extraction_delay = 300
+			var/living_players
+			switch(user_role)
+				if("Clerk")
+					extraction_delay = 50
+				if("Records Officer")	//No clue what the fuck you're doing here but go off.
+					extraction_delay = 30
+				if("Extraction Officer")
+					extraction_delay = 1
```

Comment:
30 seconds is ridiculously long.
```suggestion
			var/extraction_delay = 10 SECONDS
			var/living_players
			switch(user_role)
				if("Clerk")
					extraction_delay = 3 SECONDS
				if("Records Officer")	//No clue what the fuck you're doing here but go off.
					extraction_delay = 2 SECONDS
				if("Extraction Officer")
					extraction_delay = 0
```
Add a check for if extraction delay is correct, so we don't have to run do_after for EO.